hydra-node/src/Hydra/Node.hs

```diff
@@ -179,7 +183,7 @@ processEffect HydraNode{hn, oc, server, eq, env = Environment{party}} tracer e =
   case e of
     ClientEffect i -> sendOutput server i
     NetworkEffect msg -> broadcast hn msg >> putEvent eq (NetworkEvent msg)
-    OnChainEffect tx -> postTx oc tx
+    OnChainEffect tx -> postTx oc tx `catch` \(ex :: InvalidTxError tx) -> sendOutput server (PostTxOnChainFailed tx ex)
```

Comment:
This *should* be an event to allow the server to react to it; for now, the only sensible reaction would be sending an output to the client (like here) but later on, have extra error handling logic, retrying etc..